### ISSUE TITLE
Sections don't inherit context

### ISSUE THREAD

**Issue by KingDuckZ** [cierelabs/boostache#12]
Given this input on mustache's online demo:

    {{category}}
    {{#items}}
      {{#first}}
        <li>{{category}}: <strong>{{name}}</strong></li>
      {{/first}}
      {{#link}}
        <li>{{category}}: <a href="{{url}}">{{name}}</a></li>
      {{/link}}
    {{/items}}

and

    {
      "category": "Color",
      "items": [
          {"name": "red", "first": true, "url": "#Red"},
          {"name": "green", "link": true, "url": "#Green"},
          {"name": "blue", "link": true, "url": "#Blue"}
      ]
    }

I get this output:

    Color
    <li>Color: <strong>red</strong></li>
    <li>Color: <a href="#Green">green</a></li>
    <li>Color: <a href="#Blue">blue</a></li>

meaning that while in a subsection, parent contextes are inherited. From the tests I ran on example3.cpp (I might have done something wrong), I see this is not the case with Boostache.
This behaviour is not very clear from mustache's man page, but the online demo does lookup parent contextes.

**Comment by mjcaisse** [comment]
Indeed, this is a known issue and we are working on a more complex "context" wrapper design that allows moving up the context tree to resolve parent things.
